haystack/modeling/model/multimodal/transformers.py

```diff
@@ -0,0 +1,256 @@
+        """
+        inputs = self.get_features(data=data, **kwargs)
+
+        # Check if the inputs are correct TODO verify, does this check still makes sense to keep?
+        mandatory_args, optional_args = self.expected_inputs
+        all_args = mandatory_args | optional_args
+        given_args = set(inputs.keys())
+        if given_args < mandatory_args or given_args > all_args:
+            raise ModelingError(
+                "The features extracted from the data do not match the model's expectations.\n"
+                f"Input names: {', '.join(sorted(inputs.keys()))}\n"
+                f"Expected: {', '.join(sorted(all_args))} (where {', '.join(sorted(mandatory_args))} are mandatory)"
+            )
+
+        # Convert to tensor if necessary
+        for param_name, param_value in inputs.items():
+            if isinstance(param_value, list):
+                inputs[param_name] = torch.tensor(param_value)
+
+        # The actual encoding step
+        with torch.no_grad():
+            output = self._forward(**inputs)
+        return output
+
+    def _forward(self, **kwargs) -> torch.Tensor:
+        """
+        The default forward() implementation. Simply returns the pooler_output field of the output of the model's
+        forward pass, using the external pooler if the model does not provide one.
+        """
+        output = self.model(**kwargs)
+        if hasattr(self, "pooler"):
+            return self.pooler(output[0])
+        return output.pooler_output
+
+
+class HaystackTextTransformerModel(HaystackTransformerModel):
+    """
+    A class wrapping `transformers` models that handle text data.
+
+    Models using this wrapper should accept "input_ids", "token_type_ids", "attention_mask" as
+    input vector for their forward pass.
+    """
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str,
+        model_type: str,
+        content_type: str = "text",  # replace with ContentTypes starting Python3.8
+        model_kwargs: Optional[Dict[str, Any]] = None,
+        feature_extractor_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        if content_type != "text":
+            raise ModelingError(
+                f"{pretrained_model_name_or_path} (type {model_type}) can't handle "
+                f"data of type {content_type} or is not supported by Haystack."
+            )
+
+        super().__init__(
+            pretrained_model_name_or_path=pretrained_model_name_or_path,
+            model_type=model_type,
+            content_type="text",
+            model_kwargs=model_kwargs,
+            feature_extractor_kwargs=feature_extractor_kwargs,
+        )
+
+    @property
+    def expected_inputs(self) -> Tuple[Set[str], Set[str]]:
+        return {"input_ids", "token_type_ids", "attention_mask"}, set()
+
+    @property
+    def embedding_dim(self):
+        return self.dim
+
+
+class HaystackImageTransformerModel(HaystackTransformerModel):
```

Comment:
When is this class actually used? In the `HUGGINGFACE_TO_HAYSTACK` dict in l. 19 of `multimodal/__init__.py`, no model type is mapped to `HaystackImageTransformerModel`.

Reply:
Indeed. Let's take it out, there's no use for it right now.